File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/FontCache.java
package ro.danserboi.quotesformindandsoul;

import android.graphics.Typeface;

import java.util.Hashtable;

public class FontCache {

    private static final Hashtable<String, Typeface> fontCache = new Hashtable<>();

    public static Typeface get(String name) {
        Typeface tf = fontCache.get(name);
        if (tf == null) {
            try {
                tf = Typeface.create(name, Typeface.NORMAL);
            } catch (Exception e) {
                return null;
            }
            fontCache.put(name, tf);
        }
        return tf;
    }

}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/FontCache.kt
package ro.danserboi.quotesformindandsoul

import android.graphics.Typeface
import java.lang.Exception
import java.util.*

object FontCache {
    private val fontCache = Hashtable<String, Typeface?>()
    operator fun get(name: String): Typeface? {
        var tf: Typeface? = fontCache[name]
        if (tf == null) {
            tf = try {
                Typeface.create(name, Typeface.NORMAL)
            } catch (e: Exception) {
                return null
            }
            fontCache[name] = tf
        }
        return tf
    }
}